ext/sysvshm/sysvshm.c

```diff
@@ -203,21 +249,20 @@ PHP_FUNCTION(shm_put_var)
 	smart_str shm_var = {0};
 	php_serialize_data_t var_hash;
 
-	if (SUCCESS != zend_parse_parameters(ZEND_NUM_ARGS(), "rlz", &shm_id, &shm_key, &arg_var)) {
+	if (SUCCESS != zend_parse_parameters(ZEND_NUM_ARGS(), "Olz", &shm_id, sysvshm_ce, &shm_key, &arg_var)) {
 		RETURN_THROWS();
 	}
 
+	shm_list_ptr = Z_SYSVSHM_P(shm_id);
+	if (!shm_list_ptr->ptr) {
```

Comment:
Shouldn't these cases throw? Similar to using functions on a closed resource.

Reply:
You're right, throwing an exception would be the consistent thing to do. I think I was just too shy here to apply this practice. :) Hopefully, the error message is ok.